core/drivers/stm32_tamp.c

```diff
@@ -98,46 +111,259 @@
+ */
+struct stm32_tamp_instance {
+	struct stm32_tamp_platdata pdata;
+	uint32_t hwconf1;
+	uint32_t hwconf2;
+};
+
 /* Expects at most a single instance */
-static struct stm32_tamp_instance *stm32_tamp_device;
+static struct stm32_tamp_instance *stm32_tamp_dev;
+
+static void apply_rif_config(void)
+{
+	struct rif_conf_data *rif_conf = stm32_tamp_dev->pdata.conf_data;
+	vaddr_t base = io_pa_or_va(&stm32_tamp_dev->pdata.base, 1);
+	uint32_t access_mask_priv_reg = 0;
+	uint32_t access_mask_sec_reg = 0;
+	uint32_t privcfgr = 0;
+	uint32_t seccfgr = 0;
+	unsigned int i = 0;
+
+	if (!stm32_tamp_dev->pdata.conf_data)
+		return;
+
+	/* Build access masks for _TAMP_PRIVCFGR and _TAMP_SECCFGR */
+	for (i = 0; i < TAMP_RIF_RESOURCES; i++) {
+		if (BIT(i) & rif_conf->access_mask[0]) {
+			switch (i) {
+			case 0:
+				access_mask_sec_reg |= _TAMP_SECCFGR_TAMPSEC;
+				access_mask_priv_reg |= _TAMP_PRIVCFG_TAMPPRIV;
+				break;
+			case 1:
+				access_mask_sec_reg |= _TAMP_SECCFGR_CNT1SEC;
+				access_mask_priv_reg |= _TAMP_PRIVCFG_CNT1PRIV;
+				access_mask_priv_reg |= _TAMP_PRIVCFG_BKPRWPRIV;
+				break;
+			case 2:
+				access_mask_sec_reg |= _TAMP_SECCFGR_CNT2SEC;
+				access_mask_priv_reg |= _TAMP_PRIVCFG_CNT2PRIV;
+				access_mask_priv_reg |= _TAMP_PRIVCFG_BKPWPRIV;
+				break;
+			default:
+				panic();
+			}
+		}
+	}
+
+	/*
+	 * When TDCID, OP-TEE should be the one to set the CID filtering
+	 * configuration. Clearing previous configuration prevents
+	 * undesired events during the only legitimate configuration.
+	 */
+	if (stm32_tamp_dev->pdata.is_tdcid) {
+		for (i = 0; i < TAMP_RIF_RESOURCES; i++)
+			if (BIT(i) & rif_conf->access_mask[0])
+				io_clrbits32(base + _TAMP_CIDCFGR(i),
+					     _TAMP_CIDCFGR_CONF_MASK);
+	}
+
+	if (rif_conf->sec_conf[0] & _TAMP_SECCFGR_RIF_TAMP_SEC)
+		seccfgr |= _TAMP_SECCFGR_TAMPSEC;
+	if (rif_conf->sec_conf[0] & _TAMP_SECCFGR_RIF_COUNT_1)
+		seccfgr |= _TAMP_SECCFGR_CNT1SEC;
+	if (rif_conf->sec_conf[0] & _TAMP_SECCFGR_RIF_COUNT_2)
+		seccfgr |= _TAMP_SECCFGR_CNT2SEC;
+
+	if (rif_conf->priv_conf[0] & _TAMP_PRIVCFGR_RIF_TAMP_PRIV)
+		privcfgr |= _TAMP_PRIVCFG_TAMPPRIV;
+	if (rif_conf->priv_conf[0] & _TAMP_PRIVCFGR_RIF_R1)
+		privcfgr |= _TAMP_PRIVCFG_CNT1PRIV | _TAMP_PRIVCFG_BKPRWPRIV;
+	if (rif_conf->priv_conf[0] & _TAMP_PRIVCFGR_RIF_R2)
+		privcfgr |= _TAMP_PRIVCFG_CNT2PRIV | _TAMP_PRIVCFG_BKPWPRIV;
+
+	/* Security and privilege RIF configuration */
+	io_clrsetbits32(base + _TAMP_PRIVCFGR, access_mask_priv_reg, privcfgr);
+	io_clrsetbits32(base + _TAMP_SECCFGR, access_mask_sec_reg, seccfgr);
+
+	if (!stm32_tamp_dev->pdata.is_tdcid)
+		return;
+
+	for (i = 0; i < TAMP_RIF_RESOURCES; i++) {
+		if (!(BIT(i) & rif_conf->access_mask[0]))
+			continue;
+
+		io_clrsetbits32(base + _TAMP_CIDCFGR(i),
+				_TAMP_CIDCFGR_CONF_MASK,
+				rif_conf->cid_confs[i]);
+	}
+}
+
+static TEE_Result stm32_tamp_apply_bkpr_rif_conf(void)
+{
+	struct stm32_bkpregs_conf_new *bkpregs_conf =
+			stm32_tamp_dev->pdata.bkpregs_conf;
```

Comment:
Fix commit "drivers: stm32_tamp: add stm32mp25 support for RIF configuration":  copy the data or use a pointer? (no functional change)
```patch
 	struct stm32_bkpregs_conf_new *bkpregs_conf =
-			stm32_tamp_dev->pdata.bkpregs_conf;
+			&stm32_tamp_dev->pdata.bkpregs_conf;
```
Ditto for commit "drivers: stm32_tamp: configure the backup registers when driver is probing".

Can remove test at line 314.